builtin/logical/transit/lock_manager.go

```diff
@@ -0,0 +1,337 @@
+package transit
+
+import (
+	"encoding/json"
+	"fmt"
+	"sync"
+
+	"github.com/hashicorp/vault/logical"
+)
+
+const (
+	shared    = false
+	exclusive = true
+)
+
+type lockManager struct {
+	// A lock for each named key
+	locks map[string]*sync.RWMutex
+
+	// A mutex for the map itself
+	lockMutex sync.RWMutex
+
+	// If caching is enabled, the map of name to in-memory policy cache
+	cache map[string]*Policy
+
+	// Used for global locking, and as the cache map mutex
+	globalMutex sync.RWMutex
+}
+
+func newLockManager(cacheDisabled bool) *lockManager {
+	lm := &lockManager{
+		locks: map[string]*sync.RWMutex{},
+	}
+	if !cacheDisabled {
+		lm.cache = map[string]*Policy{}
+	}
+	return lm
+}
+
+func (lm *lockManager) CacheActive() bool {
+	return lm.cache != nil
+}
+
+func (lm *lockManager) LockAll(name string) {
```

Comment:
To reduce the surface area of the API to its minimum and to reduce the possible error of misuse from exported functions (e.g. no one outside of the transit backend should lock everything).  If it's necessary to expose down the road, then it's necessary to export, but there weren't any consumers outside of `builtin/logical/transit/lock_manager.go`.